dns/ddclient/src/opnsense/www/js/widgets/Dyndns.js

```diff
@@ -0,0 +1,136 @@
+ * OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import BaseTableWidget from "./BaseTableWidget.js";
+
+export default class Dyndns extends BaseTableWidget {
+    constructor() {
+        super();
+    }
+
+    getGridOptions() {
+        return {
+            // Trigger overflow-y:scroll after 650px height
+            sizeToContent: 650
+        };
+    }
+
+    getMarkup() {
+        let $container = $('<div></div>');
+        let $dyndnsTable = this.createTable('dyndnsTable', {
+            headerPosition: 'top',
+            headers: [
+                this.translations.service,
+                // this.translations.domains
+            ]
+        });
+
+        $container.append($dyndnsTable);
+        return $container;
+    }
+
+    async onWidgetTick() {
+        // Check if DynDNS is enabled
+        const statusData = await this.ajaxCall('/api/dyndns/service/status');
+        if (!statusData || statusData.status !== "running") {
+            this.displayError(this.translations.servicedisabled);
+            return;
+        }
+
+        // Fetch DynDNS account information
+        const accountData = await this.ajaxCall('/api/dyndns/accounts/search_item');
+        if (!accountData || !accountData.rows || accountData.rows.length === 0) {
+            this.displayError(this.translations.noaccount);
+            return;
+        }
+
+        this.processAccounts(accountData.rows);
+    }
+
+    // Utility function to display errors within the widget
+    displayError(message) {
+        const $error = $(`
+            <div class="error-message">
+                <a href="/ui/dyndns/" target="_blank">${message}</a>
+            </div>
+        `);
+        $('#dyndnsTable').empty().append($error);
+    }
+
+    processAccounts(accounts) {
+        if (!this.dataChanged('accounts', accounts)) {
+            return;
+        }
+
+        $('.dyndns-tooltip').tooltip('hide');
+
+        let rows = [];
+        accounts.forEach(account => {
+            let colorClass = account.enabled === "1" ? 'text-success' : 'text-danger';
+            let tooltipText = account.enabled === "1" ? this.translations.enabled : this.translations.disabled;
+
+            let domainNames = account.hostnames.split(',')
+                  .map(domain => `<div>${domain}</div>`)
+                  .join('');
+
+            // Convert time to a localized format
+            let localizedTime = account.current_mtime ? new Date(account.current_mtime).toLocaleString() : '-';
+            let currentIp = account.current_ip || '-';
```

Comment:
For style reasons I prefer "N/A" whenever some value is expected but **n**ot **a**vailable, which also needs to be translated.

Reply:
https://github.com/opnsense/plugins/blob/cab23a684b902e5af00815a06bba7841a62d80a9/security/etpro-telemetry/src/opnsense/www/js/widgets/ETProTelemetry.js#L52

Reply:
let's make it a thing ;)

https://github.com/opnsense/plugins/blob/cab23a684b902e5af00815a06bba7841a62d80a9/dns/rfc2136/src/www/widgets/widgets/rfc2136.widget.php#L72

Reply:
Sure np, I can fix it in the telemetry widget too, later. Gonna create an issue later.